### DISCUSSION TITLE
Heat from fire calculation

### DISCUSSION THREAD

**Discussion by TorleifVeen** [equinor/neqsim#390] [General]
Hi

A great project. We are interested in calculating depressurization with a global fire heat load. For a separator the code should work if an external heat source is added in the calculation of deltaEnergy in the runTransient function for separator? 
It is correct that heat from exterior is not included in the code of the separator itself, only pipings?

The calculation of the external heat would then be similar to the calculation for piping surroundings where temperature and heat transfer coefficient is included, but heat from radiation is not.

Are there some  details buried in the code I should be aware of that would make this not work or some other place I have to modify as well to make this work?

Regards

Torleif Veen

**Comment by EvenSol** [comment]
Hi.
I think it is correct what you say - that external sources could be added to deltaEnergy, and that heat from surroundings is not added  in the model as it is. I think you have understood the code correct. Improved calculations of blow down and fire is however something that should be tested and improved in NeqSim, and I hope someone could look into this in the future.

A relevant student work from NTNU can be [found here](https://ntnuopen.ntnu.no/ntnu-xmlui/bitstream/handle/11250/2408935/15728_FULLTEXT.pdf?sequence=1&isAllowed=y), and similar mathematics could be used in NeqSim.

Some example code should be added in the process safety part in [examples in NeqSim](https://colab.research.google.com/github/EvenSol/NeqSim-Colab/blob/master/notebooks/examples_of_NeqSim_in_Colab.ipynb).
Even
